firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AndroidConnectivityMonitor.java

```diff
@@ -99,15 +101,19 @@ public void run() {
   private class DefaultNetworkCallback extends ConnectivityManager.NetworkCallback {
     @Override
     public void onAvailable(Network network) {
-      for (Consumer<NetworkStatus> callback : callbacks) {
-        callback.accept(NetworkStatus.REACHABLE);
+      synchronized (callbacks) {
+        for (Consumer<NetworkStatus> callback : callbacks) {
+          callback.accept(NetworkStatus.REACHABLE);
+        }
       }
     }
 
     @Override
     public void onLost(Network network) {
-      for (Consumer<NetworkStatus> callback : callbacks) {
-        callback.accept(NetworkStatus.UNREACHABLE);
+      synchronized (callbacks) {
```

Comment:
I can't comment there so consider this a comment on line 133.

Don't we also need to synchronize in `NetworkReceiver` for pre-N devices?

Reply:
Yes, absolutely. Fixed/thanks.